diff --git a/DesktopEditor/freetype-2.10.4/builds/unix/ftsystem.c b/DesktopEditor/freetype-2.10.4/builds/unix/ftsystem.c
index b4d71d40e3..9c27061539 100644
--- a/DesktopEditor/freetype-2.10.4/builds/unix/ftsystem.c
+++ b/DesktopEditor/freetype-2.10.4/builds/unix/ftsystem.c
@@ -64,6 +64,11 @@
 #include <fcntl.h>
 #endif
 
+#ifdef BUILDING_WASM_MODULE
+#ifdef __linux__
+#include <unistd.h>
+#endif
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/DesktopEditor/freetype-2.10.4/src/base/ftobjs.c b/DesktopEditor/freetype-2.10.4/src/base/ftobjs.c
index c060bbbc87..3a3f52427f 100644
--- a/DesktopEditor/freetype-2.10.4/src/base/ftobjs.c
+++ b/DesktopEditor/freetype-2.10.4/src/base/ftobjs.c
@@ -16,6 +16,9 @@
  */
 
 
+#ifdef BUILDING_WASM_MODULE
+#include <wasm_jmp.h>
+#endif
 #include <freetype/ftlist.h>
 #include <freetype/ftoutln.h>
 #include <freetype/ftfntfmt.h>
@@ -166,7 +169,11 @@
 
     /* throw away volatileness; use `jump_buffer' or the  */
     /* compiler may warn about an unused local variable   */
+#ifdef BUILDING_WASM_MODULE
+    ft_longjmp_ex( *(ft_jmp_buf*) jump_buffer, 1 );
+#else
     ft_longjmp( *(ft_jmp_buf*) jump_buffer, 1 );
+#endif
   }
 
 
diff --git a/DesktopEditor/freetype-2.10.4/src/sfnt/ttcmap.c b/DesktopEditor/freetype-2.10.4/src/sfnt/ttcmap.c
index 556a712199..8aa035669e 100644
--- a/DesktopEditor/freetype-2.10.4/src/sfnt/ttcmap.c
+++ b/DesktopEditor/freetype-2.10.4/src/sfnt/ttcmap.c
@@ -3825,11 +3825,20 @@
 
             valid.num_glyphs = (FT_UInt)face->max_profile.numGlyphs;
 
+#ifdef BUILDING_WASM_MODULE
+            try
+#else
             if ( ft_setjmp( FT_VALIDATOR( &valid )->jump_buffer) == 0 )
+#endif
             {
               /* validate this cmap sub-table */
               error = clazz->validate( cmap, FT_VALIDATOR( &valid ) );
             }
+#ifdef BUILDING_WASM_MODULE
+            catch (int jmp)
+            {              
+            }
+#endif
 
             if ( !valid.validator.error )
             {
diff --git a/DesktopEditor/freetype-2.10.4/src/smooth/ftgrays.c b/DesktopEditor/freetype-2.10.4/src/smooth/ftgrays.c
index 681900fd40..34dace1e9a 100644
--- a/DesktopEditor/freetype-2.10.4/src/smooth/ftgrays.c
+++ b/DesktopEditor/freetype-2.10.4/src/smooth/ftgrays.c
@@ -97,7 +97,9 @@
    */
 #undef  FT_COMPONENT
 #define FT_COMPONENT  smooth
-
+#ifdef BUILDING_WASM_MODULE
+#include <wasm_jmp.h>
+#endif
 
 #ifdef STANDALONE_
 
@@ -542,7 +544,11 @@ typedef ptrdiff_t  FT_PtrDist;
     }
 
     if ( ras.num_cells >= ras.max_cells )
+#ifdef BUILDING_WASM_MODULE
+      ft_longjmp_ex( ras.jump_buffer, 1 );
+#else
       ft_longjmp( ras.jump_buffer, 1 );
+#endif
 
     /* insert new cell */
     cell        = ras.cells + ras.num_cells++;
@@ -1636,8 +1642,11 @@ typedef ptrdiff_t  FT_PtrDist;
   {
     int  error;
 
-
+#ifdef BUILDING_WASM_MODULE
+    try
+#else
     if ( ft_setjmp( ras.jump_buffer ) == 0 )
+#endif
     {
       if ( continued )
         FT_Trace_Disable();
@@ -1654,7 +1663,11 @@ typedef ptrdiff_t  FT_PtrDist;
                   ras.num_cells,
                   ras.num_cells == 1 ? "" : "s" ));
     }
+#ifdef BUILDING_WASM_MODULE
+    catch(int jmp)
+#else
     else
+#endif
     {
       error = FT_THROW( Memory_Overflow );
 
